pkg/pidfile/pidfile.go

```diff
@@ -0,0 +1,50 @@
+package pidfile
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strconv"
+)
+
+type PidFile struct {
+	path string
+}
+
+func checkPidFileAlreadyExists(path string) error {
+	if pidString, err := ioutil.ReadFile(path); err == nil {
+		if pid, err := strconv.Atoi(string(pidString)); err == nil {
+			if _, err := os.FindProcess(pid); err == nil {
```

Comment:
the original way of checking `/proc/pid` is probably better. `FindProcess` doesnt really **find** a process in unix system, you would still have to check if that process is alive/running.